integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/NestedForEachTestCase.java

```diff
@@ -111,9 +110,11 @@ public void testNestedForEachMediatorWithIterate() throws Exception {
             }
         }
         Assert.assertTrue(carbonLogReader.checkForLog("foreach = outer", DEFAULT_TIMEOUT, 10),
-                "Count of messages entered outer ForEach scope is incorrect");
+                "Count of messages entered outer ForEach scope is incorrect. " +
+                        "Found " + carbonLogReader.getNumberOfOccurencesForLog("foreach = outer") + " occurrences");
```

Comment:
Shall we do this inside the method, carbonLogReader.checkForLog(...)? The reason is, this is not the only test case that is failing this way and we will have to add the message to all the tests unless we add it somewhere common.

Reply:
Added a Log to checkForLog method as well.